packages/location-field/src/types.ts

```diff
@@ -0,0 +1,377 @@
+import React from "react";
+// eslint-disable-next-line prettier/prettier
+import type { IntlShape } from "react-intl";
+
+export type LocationType = "from" | "to";
+export type ResultType =
+  | "CURRENT_LOCATION"
+  | "GEOCODE"
+  | "SAVED"
+  | "SESSION"
+  | "STOP";
+export type Location = {
+  id?: string;
+  lat?: number;
+  lon?: number;
+  name?: string;
+  main?: string;
+  secondary?: string;
+};
+
+export interface LocationFieldProps {
+  /**
+   * Dispatched upon selecting a geocoded result
+   * Provides an argument in the format:
+   * ```js
+   * { location: geocodedLocation }
+   * ```
+   */
+  addLocationSearch?: ({ location: GeocodedLocation }) => void;
+  /**
+   * Determines whether the input field of this component should auto-focus on first display.
+   */
+  autoFocus?: boolean;
+  /**
+   * Used for additional styling with styled components for example.
+   */
+  className?: string;
+  /**
+   * Dispatched whenever the clear location button is clicked.
+   * Provides an argument in the format:
+   *
+   * ```js
+   * { locationType: string }
+   * ```
+   */
+  clearLocation?: ({ locationType: LocationType }) => void;
+  /**
+   * The current position of the user if it is available.
+   */
+  currentPosition?: {
+    coords?: {
+      latitude: number;
+      longitude: number;
+    };
+    error?: { message: string };
+    fetching?: boolean;
+  };
+  /**
+   * A slot for the icon to display for the current position
+   */
+  currentPositionIcon?: React.ReactNode;
+  /**
+   * A slot for the icon to display for when the current position is unavailable
+   */
+  currentPositionUnavailableIcon?: React.ReactNode;
+  /**
+   * Allows the component to be rendered with pre-filled results
+   */
+  initialSearchResults?: {
+    type?: string;
```

Comment:
Sort attributes

Reply:
Addressed in 18d5606109e7dfce7d00d42b42c8b1b6f075a0bc